pkg/sdk/application_packages_def.go

```diff
@@ -0,0 +1,160 @@
+
+var applicationPackageUnsetReleaseDirective = g.NewQueryStruct("UnsetReleaseDirective").
+	Text("ReleaseDirective", g.KeywordOptions().NoQuotes().Required())
+
+var applicationPackageSetDefaultReleaseDirective = g.NewQueryStruct("SetDefaultReleaseDirective").
+	TextAssignment("VERSION", g.ParameterOptions().NoQuotes().Required()).
+	NumberAssignment("PATCH", g.ParameterOptions().NoQuotes().Required())
+
+var applicationPackageAddVersion = g.NewQueryStruct("AddVersion").
+	OptionalText("VersionIdentifier", g.KeywordOptions().NoQuotes()).
+	TextAssignment("USING", g.ParameterOptions().NoEquals().SingleQuotes().Required()).
+	OptionalTextAssignment("Label", g.ParameterOptions().SingleQuotes())
+
+var applicationPackageDropVersion = g.NewQueryStruct("DropVersion").
+	Text("VersionIdentifier", g.KeywordOptions().NoQuotes().Required())
+
+var applicationPackageAddPatchForVersion = g.NewQueryStruct("AddPatchForVersion").
+	OptionalText("VersionIdentifier", g.KeywordOptions().NoQuotes().Required()).
+	TextAssignment("USING", g.ParameterOptions().NoEquals().SingleQuotes().Required()).
+	OptionalTextAssignment("Label", g.ParameterOptions().SingleQuotes())
+
+var applicationPackageSet = g.NewQueryStruct("ApplicationPackageSet").
+	OptionalNumberAssignment("DATA_RETENTION_TIME_IN_DAYS", g.ParameterOptions().NoQuotes()).
+	OptionalNumberAssignment("MAX_DATA_EXTENSION_TIME_IN_DAYS", g.ParameterOptions().NoQuotes()).
+	OptionalTextAssignment("DEFAULT_DDL_COLLATION", g.ParameterOptions().SingleQuotes()).
+	OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes()).
+	PredefinedQueryStructField("Distribution", "*Distribution", g.ParameterOptions().SQL("DISTRIBUTION"))
+
+var ApplicationPackagesDef = g.NewInterface(
+	"ApplicationPackages",
+	"ApplicationPackage",
+	g.KindOfT[AccountObjectIdentifier](),
+).CreateOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/create-application-package",
+	g.NewQueryStruct("CreateApplicationPackage").
+		Create().
+		SQL("APPLICATION PACKAGE").
+		IfNotExists().
+		Name().
+		OptionalNumberAssignment("DATA_RETENTION_TIME_IN_DAYS", g.ParameterOptions().NoQuotes()).
+		OptionalNumberAssignment("MAX_DATA_EXTENSION_TIME_IN_DAYS", g.ParameterOptions().NoQuotes()).
+		OptionalTextAssignment("DEFAULT_DDL_COLLATION", g.ParameterOptions().SingleQuotes()).
+		OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes()).
+		PredefinedQueryStructField("Distribution", "*Distribution", g.ParameterOptions().SQL("DISTRIBUTION")).
+		OptionalTags().
+		WithValidation(g.ValidIdentifier, "name"),
+).AlterOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/alter-application-package",
+	g.NewQueryStruct("AlterApplicationPackage").
+		Alter().
+		SQL("APPLICATION PACKAGE").
+		IfExists().
+		Name().
+		OptionalQueryStructField(
+			"Set",
+			applicationPackageSet,
+			g.KeywordOptions().SQL("SET"),
+		).
+		OptionalSQL("UNSET DATA_RETENTION_TIME_IN_DAYS").
```

Comment:
unset multiple params does not seem to be working.

Reply:
I was more wondering about the structure; that the set is contained within a struct, but unset is not. I would say it's a nit-pick, but to make set and unset consistent I would move it into "unset" struct with validation saying that only one unset can be set. It will be pretty much the same except unset operations will be groupped.